### ISSUE TITLE
Move interfaces to own files and provide type declarations

### ISSUE THREAD

**Issue by tripodsan** [adobe/helix-log#70]
The current `log.js` is quite large and contains a log of functions and classes.
IMO, at least the interfaces that are not tried to the `log` module should be refactored in their own files.

An additional bonus would be to provide typescript like type declarations.

**Comment by koraa** [comment]
Good idea! I have tried splitting log.js multiple times and ran into circular dependency issues.
Maybe you will have more luck!

I think typescript declarations would be a good idea too; should be a separate issue though unless you *really* want to tackle both at once. If we do that we might provide proper type script interfaces right away…
